### PR TITLE
ipatests: refactor and extend tests for IPA-Samba integration

### PR DESCRIPTION
**Pull request by wladich**
Add tests for following scenarios:
    * running `ipa-client-samba --uninstall` without prior installation
    * mount and access Samba share by IPA user
    * mount and access Samba share by AD user
    * mount samba share by one IPA user and access it by another one
    * try mount samba share without kerberos authentication
    * uninstall and reinstall ipa-client-samba
    
  Relates: https://pagure.io/freeipa/issue/3999

### CODE REVIEW

File: ipatests/pytest_ipa/integration/host.py [hunk 1]
```diff
@@ -106,14 +106,23 @@ def to_env(self, **kwargs):
 
     def run_command(self, argv, set_env=True, stdin_text=None,
                     log_stdout=True, raiseonerr=True,
-                    cwd=None, bg=False, encoding='utf-8'):
-        # Wrap run_command to log stderr on raiseonerr=True
+                    cwd=None, bg=False, encoding='utf-8', ok_returncode=0):
+        """
```
Comment (miskopo) [inline ipatests/pytest_ipa/integration/host.py]: PEP257 violation, please, use """ and docstring inline

**Comment by Tiboris** [comment]
@wladich nitpick: please always have some brief description for PR.

**Comment by wladich** [comment]
Added description of new tests in commit message

**Comment by netoarmando** [comment]
@wladich I'm investigating the PR-CI failure. It happened in the past but I haven't seen it in a long time.

**Comment by wladich** [comment]
Fixed pr-ci definitions for nightly-f29 and nightly-rawhide

**Comment by abbra** [comment]
LGTM. All tests passed.

**Comment by wladich** [comment]
master:

* 074bf285f185d96d67cf9f410f1e8935078d15eb ipatests: allow to pass additional options for clients installation
* 3fa7865ff8c48a35d0d120c74da76ea4076a6aa4 ipatests: add utility functions related to using and managing user accounts
* 1fe69f352b28c7bbf218c9d3ece4b45eac6ddad6 ipatests: modify run_command to allow specify successful return codes
* 1d033b040d8cc96d7c6dfa0786fc09505b3a9acf ipatests: refactor and extend tests for IPA-Samba integration

**Pull request closed without merging**